File: Implementacion/Codigo/Frontend/src/app/interfaces/product.ts
export interface Product {
    id: number;
    code: string;
    name: string;
    description: string;
    price: number;
    quantity: number;
    pharmacy: {
      id: number;
      name: string;  
    };
  }

  export class ProductRequest {
    name: string;
    description: string;
    price: number;

    constructor(name: string, description: string, price: number){
      this.name = name;
      this.description = description;
      this.price = price;
    }
  }

File: Implementacion/Codigo/Backend/PharmaGo.Domain/Entities/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PharmaGo.Domain.Entities
{
    public class Product
    {
        public Product()
        {
        }

        public int Id { get; set; }
        public string Code { get; set; }
        public string Name { get; set; }
        public decimal Price { get; set; }
        public bool Deleted { get; set; }
        public string Description { get; set; }
        public Pharmacy? Pharmacy { get; set; }
    }
}
